api/desecapi/management/commands/chores.py

```diff
@@ -46,7 +48,44 @@ def update_healthcheck_timestamp():
             serializer.save()
         print(f'TXT {name} updated to {content}')
 
+    @staticmethod
+    def alerting_healthcheck():
+        name = 'external-timestamp.desec.test'
+        try:
+            models.Domain.objects.get(name=name)
+        except models.Domain.DoesNotExist:
+            print(f'{name} zone is not configured; skipping alerting health check')
+            return
+
+        now = time.time()
+        timestamps = []
+        query = dns.message.make_query(f'{name}.', dns.rdatatype.TXT)
+        server = 'ns1.desec.io'
+        response = None
+        try:
+            response = dns.query.tcp(query, server, timeout=5)
+            for rrset in response.answer:
+                if rrset.rdtype != dns.rdatatype.TXT:
+                    continue
+                for txt_content in rrset:
+                    timestamps.append(str(txt_content)[1:-1])
+
+            if any(now - 600 <= int(timestamp) <= now for timestamp in timestamps):
+                print(f'TXT {name} up to date.')
+                return
+        except Exception:
+            pass
+
+        print(f'TXT {name} out of date!')
+        subject = 'ALERT Alerting system down?'
+        message = f'TXT query for {name} on {server} gave the following response:\n'
+        message += f'{str(response)}\n\n'
+        message += 'Extracted timestamps in TXT RRset:\n'
+        message += str(timestamps)
+        mail_admins(subject, message, connection=get_connection('django.core.mail.backends.smtp.EmailBackend'))
+
     def handle(self, *args, **kwargs):
         self.delete_expired_captchas()
         self.delete_never_activated_users()
         self.update_healthcheck_timestamp()
+        self.alerting_healthcheck()
```

Comment:
maybe in handle we should wrap each call into a separate try/except? Now if something above raises uncaught exceptions, then the health check will not execute

Reply:
Or just do the critical things first :)

Reply:
reordered and wrapped the whole thing in try-except so that we get an email when anything goes wrong